src/JSON.jl

```diff
@@ -41,8 +39,8 @@ suffix(::State{NOINDENT}) = ""
 prefix(s::State{INDENT})  = s.prefix
 prefix(::State{NOINDENT}) = ""
 
-colon(s::State{INDENT})  = ": "
-colon(::State{NOINDENT}) = ":"
+separator(s::State{INDENT})  = ": "
```

Comment:
You can remove the `s` here.  This was probably copy and pasted from `prefix` above (possibly by me), which does need `s` for the first definition.